cmds/clients/contestcli-http/main.go

```diff
@@ -149,12 +188,36 @@ func run(verb string) error {
 		encoder.SetIndent("", " ")
 		err := encoder.Encode(apiErr)
 		if err != nil {
-			return fmt.Errorf("cannot re-encode httplistener.HTTPAPIResponse object: %v", err)
+			return "", fmt.Errorf("cannot re-encode httplistener.HTTPAPIResponse object: %v", err)
 		}
 		indentedJSON = buffer.Bytes()
 	}
 	// this is the only thing we want on stdout - the JSON-formatted response,
 	// so it can be piped to other tools if desired.
-	fmt.Println(string(indentedJSON))
-	return nil
+	return string(indentedJSON), nil
 }
+ 
+func wait(params url.Values, jobWaitPoll time.Duration) (string, error) {
+	// keep polling for status till job is completed, used when -wait is set
+	for {
+		resp, err := request("status", params)
+		if err != nil {
+			return "", err
+		}
+
+		parsedData := &api.ResponseDataStatus{}
+		parsedResp := &httplistener.HTTPAPIResponse{Data: parsedData}
+		if err := json.Unmarshal([]byte(resp), parsedResp); err != nil {
+			return "", fmt.Errorf("cannot decode json response: %v", err)
+		}
+		jobState := parsedData.Status.State
```

Comment:
Sorry, forgot to mention this earlier:

Sometimes Status may be `nil` ([it happens in `api.Status()`](https://github.com/facebookincubator/contest/blob/ebae6008ef2a5154929cf252ccdb9ff1e24b01aa/pkg/api/api.go#L201)). So this line will cause `panic` (since there's no way to dereference `nil` to access field `State`).

And therefore it makes sense to add before this line an `if`:

```go
if parsedResp.Error != nil {
    return "", fmt.Errorf("server responded with an error: %s", *parsedResp.Error)
}
```

Reply:
Good catch.  Added  this as well.